assets/js/blocks/product-query/variations/product-query.tsx

```diff
@@ -90,8 +87,14 @@ if ( isWpVersion( '6.1', '>=' ) ) {
 			unregisterBlockVariation( QUERY_LOOP_ID, VARIATION_NAME );
 
 			registerProductsBlock( queryAttributes );
-		} );
-	} else {
-		registerProductsBlock( QUERY_DEFAULT_ATTRIBUTES );
-	}
+		}
+	}, 'core/edit-site' );
+
+	let isBlockRegistered = false;
+	subscribe( () => {
+		if ( ! isBlockRegistered ) {
+			isBlockRegistered = true;
```

Comment:
Out of curiosity, why do we need to keep track of `isBlockRegistered` in the post editor but not in the site editor?

Reply:
Oh, I saw it now, not having it causes the Site Editor to freeze.